File: backend/Models/LedgerEntry.cs
// Source: C# (Original)
using System.Text.Json.Serialization;

namespace FinanceApp.Models
{
    public class LedgerEntry
    {
        public string Id { get; set; } = Guid.CreateVersion7().ToString();
        public string UserId { get; set; } = string.Empty;
        public string TransactionId { get; set; } = string.Empty;
        public string AccountId { get; set; } = string.Empty;
        public decimal Amount { get; set; }
        public string? Note { get; set; } = "";
        public string? ReferenceNumber { get; set; }

        [JsonIgnore]
        public Transaction? Transaction { get; set; }
    }
}

File: notif-ingester/models/ledger_entry.py
from typing import Optional
from pydantic import BaseModel, Field

class LedgerEntry(BaseModel):
    id: str = Field(alias="Id")
    user_id: str = Field(alias="UserId")
    transaction_id: str = Field(alias="TransactionId")
    account_id: str = Field(alias="AccountId")
    amount: float = Field(alias="Amount")
    note: Optional[str] = Field(default=None, alias="Note")
    reference_number: Optional[str] = Field(default=None, alias="ReferenceNumber")

    class Config:
        populate_by_name = True
